horovod/_keras/__init__.py

```diff
@@ -58,40 +61,9 @@ def _aggregate_gradients(self, grads_and_vars):
                 return aggregated_grads
             return list(zip(aggregated_grads, vars))
 
-        def _allreduce(self, gradients):
+        def _allreduce(self, grads):
             self._aggregated_gradients = True
-            if hvd.size() > 1:
-                if self._gradient_predivide_factor != 1.0:
-                    # Perform averaging via pre/postscaling factors.
-                    # Split average operation across pre/postscale factors
-                    prescale_factor = 1.0 / gradient_predivide_factor
-                    postscale_factor = gradient_predivide_factor / hvd.size()
-                    do_average = False
-                else:
-                    prescale_factor = 1.0
-                    postscale_factor = 1.0
-                    do_average = True
-
-                averaged_gradients = []
-                with tf.name_scope(self._name + "_Allreduce"):
-                    for grad in gradients:
-                        if grad is not None:
-                            if self._sparse_as_dense and \
-                                    isinstance(grad, tf.IndexedSlices):
-                                grad = tf.convert_to_tensor(grad)
-                            avg_grad = hvd.allreduce(grad,
-                                                     average=do_average,
-                                                     device_dense=self._device_dense,
-                                                     device_sparse=self._device_sparse,
-                                                     compression=self._compression,
-                                                     prescale_factor=prescale_factor,
-                                                     postscale_factor=postscale_factor)
-                            averaged_gradients.append(avg_grad)
-                        else:
-                            averaged_gradients.append(None)
-                    return averaged_gradients
-            else:
-                return gradients
+            return self._allreduce_grads(grads)
```

Comment:
@romerojosh this part may be relevant to you since you recently did some work on this part of the code.

Reply:
Excellent! Glad to see this duplicate code being removed.